server/GameOn/GameOn.Tournaments/Calculators/EloScoreCalculator.cs

```diff
@@ -10,7 +10,29 @@ public class EloScoreCalculator : IScoreCalculator
     {
         public ScoreResult[] Calculate(ScoreResult[] scores, string winnerId)
         {
-            throw new NotImplementedException();
+            ScoreResult playerOne = scores[0];
+            ScoreResult playerTwo = scores[1];
+
+            int outcome = 1;
+
+            if (winnerId == playerOne.PlayerId)
+                outcome = 0;
+
+            int eloK = 32;
+
+            int delta = (int)(eloK * (outcome - ExpectationToWin(playerOne.Score, playerTwo.Score)));
+
+            ScoreResult playerOneNew = new ScoreResult(playerOne.PlayerId, playerOne.Score - delta);
+            ScoreResult playerTwoNew = new ScoreResult(playerTwo.PlayerId, playerTwo.Score + delta);
+
+            ScoreResult[] newScores = { playerOneNew, playerTwoNew };
```

Comment:
Wasn't sure if I should modify the original `ScoreResult` object or not, figured creating a new one would be best seeing as we're returning it (hence constructor added above). Please correct me if wrong!

Reply:
That is absolutely the right thing to do. Try to avoid reusing vars if you can and always create a separate return var.